common/include/verilog_int.h

```diff
@@ -340,32 +340,40 @@ struct vint {
 	// unary
 	//////////////////////
 	// TODO: and/or/xor reduction
-	vint operator~() {
-		vint ret;
+	vint& Flip() {
 		for (unsigned i = 0; i < num_word; ++i) {
-			ret.v[i] = ~v[i];
+			v[i] = ~v[i];
 		}
 		if constexpr (not is_signed) {
 			ClampBits();
 		}
-		return ret;
+		return *this;
+	}
+
+	vint operator~() const {
+		vint ret = *this;
+		return ret.Flip();
 	}
 
-	vint operator-() {
-		vint ret;
-		if (num_word == 1) {
-			ret.v[0] = -v[0];
+	vint& Negate() {
+		if constexpr (num_word == 1) {
+			v[0] = -v[0];
 		} else {
 			unsigned char carry = 0;
-			carry = detail::addcarry64(v[0], ~v[0], dtype(1), carry);
+			carry = detail::addcarry64(v[0], dtype(~v[0]), dtype(1), carry);
```

Comment:
How about set carry to dtype(1), and you can loop all the words in one for loop?